### PR TITLE
Make legend toggle work in IE11

### PR DESCRIPTION
**Pull request by GDFaber**
- Replaced ```display: initial``` by ```display: block``` in Chart.prototype.show() to stop hidden data items from disappearing in IE11 when they should be shown again (as described in #2528).
- Built and tested the changes locally and run tests according to contributing guide.
- Did not add new tests; if I should do that here, please give me a pointer on how it is done in this case.

### CODE REVIEW

**Comment by kt3k** [comment]
@GDFaber Thank you for looking into this. The change looks almost good, but we'd like to minimize the effect of change. So can you add the conditional like the below instead of just changing to 'block'?

```
isIE ? 'block' : 'initial'
```

**Comment by GDFaber** [comment]
Hi @kt3k,

sure, I'll add IE detection as stated above. I have a question about that:

I found a check for IE9 in clip.js/ChartInternal.prototype.getClipPath() as well. Should I add a detection function to util.js (as a part of this PR) and let getClipPath() use it (this would be a new PR then)? or is it ok to just check locally and leave clip.js as it is?

**Comment by kt3k** [comment]
> Should I add a detection function to util.js (as a part of this PR) and let getClipPath() use it (this would be a new PR then)?

That sounds good to me.

**Comment by GDFaber** [comment]
- Added functions for IE detection in utils.js
- Added basic unit tests for the IE detection functions

File: src/util.js [hunk 1]
```diff
@@ -87,3 +87,38 @@ export var isWithinBox = function(point, box, sensitivity = 0) {
     return xStart < point[0] && point[0] < xEnd && yEnd < point[1] && point[1] < yStart;
 };
 
+/**
+ * Returns Internet Explorer version number (or false if no Internet Explorer used).
+ */
+export var getIEVersion = function() {
+    // https://stackoverflow.com/questions/19999388/check-if-user-is-using-ie
+    const agent = window.navigator.userAgent;
```
Comment (kt3k) [inline src/util.js]: How about taking `agent` as function paramter? If we do so, we can write the tests like the below:

```
expect(getIEVersion('Mozilla/5.0 (Windows NT 6.3; Trident/7.0; rv:11.0) like Gecko ')).toBe(11)
```

File: src/api.show.js [hunk 2]
```diff
@@ -10,7 +11,7 @@ Chart.prototype.show = function (targetIds, options) {
     targets = $$.svg.selectAll($$.selectorTargets(targetIds));
 
     targets.transition()
-        .style('display', 'initial', 'important')
+        .style('display', isIE() ? 'block' : 'initial', 'important')
```
Comment (kt3k) [inline src/api.show.js]: nice!

File: src/util.js (hunk 1 above)
Comment (GDFaber) [reply src/util.js]: Sure, let's do this. I'm going to modify getIEVersion() to take an optional parameter so usage is easy and can be tested as well.

**Pull request merged**